pkg/controlplane/apiserver/config.go

```diff
@@ -179,6 +177,19 @@ func BuildGenericConfig(
 	return
 }
 
+func buildStorageFactory(s controlplaneapiserver.CompletedOptions, genericConfig *genericapiserver.Config) (*serverstorage.DefaultStorageFactory, error) {
+	storageFactoryConfig := kubeapiserver.NewStorageFactoryConfig()
+	storageFactoryConfig.APIResourceConfig = genericConfig.MergedResourceConfig
+	storageFactory, err := storageFactoryConfig.Complete(s.Etcd).New()
```

Comment:
have not looked deeper. But `Complete` is for completion and I imagine one could do the defaulting there? Or even in `New()`. Why should the default happen out of band?

Reply:
@sttts `s.Etcd.StorageConfig.StorageObjectCountTracker` is nil by default. 

`Complete` copy `s.Etcd.StorageConfig` to `StorageFactoryConfig`, so `StorageFactoryConfig.StorageConfig.StorageObjectCountTracker` is nil.

 `genericapiserver.Config` has a StorageObjectCountTracker from `flowcontrolrequest.NewStorageObjectCountTracker()`, which also will be used in `CRDRESTOptionsGetter`.

Reply:
Why not set it correctly in `s.Etcd.StorageConfig`?

What I am after is that overriding it here feels fishy. It is too late. Either we set the tracker always in `storagebackend.NewDefaultConfig` or in the path that calls that func for kube-apiserver.

Reply:
@sttts Set the tracker always in `storagebackend.NewDefaultConfig`, and copy it to `genericConfig`. PTAL